toolchain/sem_ir/function.h

```diff
@@ -49,6 +50,11 @@ struct FunctionFields {
   // function, in lexical order. The first block is the entry block. This will
   // be empty for declarations that don't have a visible definition.
   llvm::SmallVector<InstBlockId> body_block_ids = {};
+
+  // If the function is imported from C++, points to the Clang declaration in
+  // the AST. Used for mangling. The AST is owned by `CompileSubcommand` so we
+  // expect it to be live from `Function` creation to mangling.
+  const clang::NamedDecl* cpp_clang_decl = nullptr;
```

Comment:
Maybe `cpp_decl` instead of `cpp_clang_decl`, similar to how we say `cpp_ast` instead of `clang_ast` or `cpp_clang_ast`? i.e., "cpp" and "clang" mean more or less the same thing since we can't support other C++ compilers this way.

```suggestion
  const clang::NamedDecl* cpp_decl = nullptr;
```

Reply:
Done.

Reply:
Also renamed `cpp_clang_mangle_context_` to `cpp_mangle_context_`.